### ISSUE TITLE
psalm-assert-if-true not working for class constants

### ISSUE THREAD

**Issue by ndench** [vimeo/psalm#3876]
Using `@psalm-assert-if-true Foo::KEY_*` on a function does not assert that the parameter is a class constant.

~https://psalm.dev/r/53bdb32e52~ https://psalm.dev/r/8a49288c4b

```php
<?php

class Foo {
    public const KEY_FOO = 'foo';
    public const KEY_BAR = 'bar';
    
    /** @psalm-assert-if-true self::KEY_* $key */
    public static function isValid(string $key): bool
    {
        return true === \in_array($key, [self::KEY_FOO, self::KEY_BAR], true);
    }
    
    /**
     * @psalm-param Foo::KEY_* $key
     */
    public static function doFoo(string $key): void
    {
    }
}

/** @var string $x */
$x = 'testing';

if (Foo::isValid($x)) {
    Foo::doFoo($x);
}
```

**Comment by weirdan** [comment]
Your snippet missed actual assert, here's a fixed (and failing) version: https://psalm.dev/r/8a49288c4b

**Comment by ndench** [comment]
Ah, right you are. Thanks @weirdan!

**Comment by ygottschalk** [comment]
duplicate of #3741

**Comment by muglug** [comment]
Fixed in 42c3a70
